src/Thinktecture.Relay.Connector/Targets/ClientRequestHandler.cs

```diff
@@ -48,66 +45,36 @@ public class ClientRequestHandler<TRequest, TResponse, TAcknowledge> : IClientRe
 			_acknowledgeTransport = acknowledgeTransport ?? throw new ArgumentNullException(nameof(acknowledgeTransport));
 
 			_acknowledgeEndpoint = new Uri($"{relayConnectorOptions.Value.DiscoveryDocument.AcknowledgeEndpoint}/");
-
-			ThreadPool.GetMinThreads(out _minWorkerThreads, out _);
-			ThreadPool.GetMaxThreads(out _maxWorkerThreads, out _maxCompletionPortThreads);
 		}
 
 		/// <inheritdoc />
-		public int? BackgroundTaskLimit
+		public Task HandleAsync(TRequest request, CancellationToken cancellationToken = default)
 		{
-			get
-			{
-				ThreadPool.GetMaxThreads(out var workerThreads, out _);
-				return workerThreads;
-			}
-			set
-			{
-				// no lower than minimum worker threads (or system's default fallback)
-				var workerThreads = Math.Max(_minWorkerThreads, value ?? _maxWorkerThreads);
-				// no lower than processor count
-				workerThreads = Math.Max(Environment.ProcessorCount, workerThreads);
-				ThreadPool.SetMaxThreads(workerThreads, _maxCompletionPortThreads);
-			}
-		}
-
-		/// <inheritdoc />
-		public async Task HandleAsync(TRequest request, CancellationToken cancellationToken = default)
-		{
-			if (QueueWorker(request, cancellationToken)) return;
-
-			if (request.AcknowledgeMode == AcknowledgeMode.ConnectorReceived)
-			{
-				await AcknowledgeRequestAsync(request, false);
-			}
-
-			await DeliverResponseAsync(request.CreateResponse<TResponse>(HttpStatusCode.ServiceUnavailable), request.EnableTracing);
+			Task.Run(() => WorkerCallAsync(request, cancellationToken), cancellationToken);
```

Comment:
```suggestion
 			if (request.AcknowledgeMode == AcknowledgeMode.ConnectorReceived)
 			{
 				await AcknowledgeRequestAsync(request, false);
 			}

			Task.Run(() => WorkerCallAsync(request, cancellationToken), cancellationToken);
```

Reply:
That would duplicate the ConnectorReceived ACK again ;)

Reply:
We would need to decide if it should be done by the handler before handing the request to the worker, or by the worker in the finally block. The difference would be that atm the ack might be delayed by the worker's timeout when requesting the target.

Reply:
Finally it is!